coriolisclient/tests/v1/test_logging.py

```diff
@@ -0,0 +1,278 @@
+            )
+
+    def test_get_endpoint_url(self):
+        self.logger._cli.get_endpoint.return_value = "url/endpoint_url/"
+
+        result = self.logger._get_endpoint_url(mock.sentinel.name)
+
+        self.assertEqual(
+            "url/endpoint_url",
+            result
+        )
+        self.logger._cli.get_endpoint.assert_called_once_with(
+            service_type=mock.sentinel.name)
+
+    def test_get_endpoint_url_not_found(self):
+        self.logger._cli.get_endpoint.return_value = None
+
+        self.assertRaises(
+            exceptions.LoggingEndpointNotFound,
+            self.logger._get_endpoint_url,
+            mock.sentinel.name
+        )
+        self.logger._cli.get_endpoint.assert_called_once_with(
+            service_type=mock.sentinel.name)
+
+    def test_get_endpoint_url_http_unauthorized(self):
+        self.logger._cli.get_endpoint.side_effect = http.Unauthorized
+
+        with self.assertLogs(logger=logging.LOG, level="ERROR"):
+            self.assertRaises(
+                exceptions.HTTPAuthError,
+                self.logger._get_endpoint_url,
+                mock.sentinel.name
+            )
+            self.logger._cli.get_endpoint.assert_called_once_with(
+                service_type=mock.sentinel.name)
+
+    @ddt.data(
+        {
+            "query_args": {
+                "arg1": None,
+                "arg2": None
+            },
+            "is_websocket": True,
+            "expected_result":
+                "ws:///None/sentinel.resource"
+        },
+        {
+            "query_args": {
+                "arg1": None,
+                "arg2": "mock_arg2"
+            },
+            "_ep_url": "https:///ep_url",
+            "is_websocket": True,
+            "expected_result":
+                "wss:///ep_url/sentinel.resource?arg2=mock_arg2"
+        },
+        {
+            "query_args": {
+                "arg1": "mock_arg1",
+                "arg2": "mock_arg2"
+            },
+            "_ep_url": "https:///ep_url",
+            "is_websocket": False,
+            "expected_result": "https:///ep_url/sentinel.resource"
+                "?arg1=mock_arg1&arg2=mock_arg2"
+        }
+    )
+    @mock.patch.object(logging.LoggingClient, '_get_endpoint_url')
+    def test_construct_url(self, data, mock_get_endpoint_url):
+        self.logger._ep_url = None
+        mock_get_endpoint_url.return_value = data.get("_ep_url", None)
+
+        result = self.logger._construct_url(
+            mock.sentinel.resource,
+            data.get("query_args"),
+            is_websocket=data.get("is_websocket", False),
+        )
+
+        self.assertEqual(
+            data.get("expected_result"),
+            result
+        )
+
+    @ddt.data(
+        (None, None, False, False),
+        ("1", 1, False, False),
+        ("1234567890123456789", None, True, True),
+        ("abc", None, True, True),
+        ("", None, True, True),
+        ("10s", mock.ANY, False, True),
```

Comment:
Put a timestamp result here please. You could achieve this by mocking `datetime.datetime.utcnow`, that way you have a predefined time, and can anticipate the resulting period timestamp.